packages/grid/x-data-grid-premium/src/hooks/features/rowGrouping/createGroupingColDef.tsx

```diff
@@ -74,11 +74,6 @@ const getLeafProperties = (leafColDef: GridColDef): Partial<GridColDef> => ({
       }
 
       return (params) => {
-        // We only want to filter leaves
-        if (params.rowNode.groupingField != null) {
-          return true;
-        }
-
```

Comment:
@flaviendelangle Filtering with grouping is hard to fully understand. I do not get the reason of all those check. 

Removing them do not break tests, so I assume filtering only on the children content leads to the same behavior